lib/get-client.js

```diff
@@ -10,5 +32,26 @@ module.exports = (githubToken, githubUrl, githubApiPathPrefix) => {
     pathPrefix: githubApiPathPrefix,
   });
   github.authenticate({type: 'token', token: githubToken});
+  retryable(github, RETRY_FUNCTIONS, retryConf);
+
   return github;
 };
+
+function retryable(obj, props, retryConf) {
+  for (const prop of props) {
+    const func = get(obj, prop);
+    set(obj, prop, (...args) =>
+      pRetry(async () => {
+        try {
+          return await func(...args);
+        } catch (err) {
+          if (SKIP_RETRY_CODES.includes(err.code)) {
+            // If the API return a 400, 401 or 403 error, do not retry
+            throw new pRetry.AbortError(err);
+          }
+          throw err;
+        }
+      }, retryConf)
+    );
+  }
+}
```

Comment:
I think this implementation is rather hard to read ... maybe see the implementation from Greenkeeper for comparison: https://github.com/greenkeeperio/greenkeeper/blob/1704eaca48389699725656242b5c2e1f9213e3b2/lib/github.js#L6-L27 (not that the `.hook` API is not public API yet so we'd need to pin the `@octokit/rest` package).

Do we need to hard-code the methods we want to implement retry? Maybe we could do retries on all writing operations + search? 

The next milestone for `@ocotkit/rest` will be all things APIs and I will look into making best practises like throttling and retries  part of `@ocotkit/rest` itself. So if you like we can merge this as is but would be good to review it once throttling is implemented in `@octokit/rest` itself

Reply:
Using the  `octokit.hook.error`seems more elegant than replacing the function in the client object like I do. I can change the code to use that. That would avoid having to list the functions we want to wrap.

Do we have to pin the exact version or just `~14.0.0`?

Once it's in `@ocotkit/rest` will review and make the changes necessary to use it.